### ISSUE TITLE
Server headers/sources generating

### ISSUE THREAD

**Issue by twaik** [NilsBrause/waylandpp#62]
Hi. Is it possible?

**Comment by NilsBrause** [comment]
You could try the server branch. I just rebased it onto master and fixed the complie errors.
However, there is no guarantee that it actually works, since I haven't really touched it in 2 years.

**Comment by NilsBrause** [comment]
Server support is now available in Version 1.0.0.

**Comment by twaik** [comment]
Hi. Did you test this? I tried to follow the pingpong example but failed to create on_destroy callback.
I built `wayland-server.cpp` and `wayland-util.cpp` with my source to get more debugging data.

```
#include <iostream>
#include <wayland-util.hpp>
#include <wayland-server.hpp>
#include <wayland-server-protocol.hpp>
using namespace wayland::server;

int main() {
	display_t display;
	
	display.on_client_created() = [=] (client_t& client) {
		std::cout << "Client created!!! " << std::endl;
		client.on_destroy() = [client] () {
			std::cout << "Client destroyed!!! " << std::endl;
		};
	};

	std::cout << "Starting server" << std::endl;

	// create wayland UNIX socket.
	display.add_socket("wayland-0");
	display.run();
	
	std::cout << "Shutting server" << std::endl;
	
	return 0;
}
```
Log + gdb log.
```
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Starting server
Client created!!! 

Program received signal SIGSEGV, Segmentation fault.
0x0000555555576db4 in std::__atomic_base<unsigned int>::fetch_sub (__m=std::memory_order_seq_cst, __i=1, this=0x50) at /usr/include/c++/11/bits/atomic_base.h:646
646	      { return __atomic_fetch_sub(&_M_i, __i, int(__m)); }
#0  0x0000555555576db4 in std::__atomic_base<unsigned int>::fetch_sub (__m=std::memory_order_seq_cst, __i=1, this=0x50) at /usr/include/c++/11/bits/atomic_base.h:646
#1  std::__atomic_base<unsigned int>::operator-- (this=0x50) at /usr/include/c++/11/bits/atomic_base.h:370
#2  0x000055555557341a in wayland::server::client_t::fini (this=0x7fffffffd8b0) at wayland-server.cpp:306
#3  0x000055555557357a in wayland::server::client_t::~client_t (this=0x7fffffffd8b0, __in_chrg=<optimized out>) at wayland-server.cpp:329
#4  0x00005555555710dc in ~<lambda>(void) (this=0x7fffffffd8b0, __in_chrg=<optimized out>) at lorie.cpp:12
#5  0x0000555555571168 in operator() (__closure=0x55555559d0b0, client=...) at lorie.cpp:14
#6  0x0000555555571b4f in std::__invoke_impl<void, main()::<lambda(wayland::server::client_t&)>&, wayland::server::client_t&>(std::__invoke_other, struct {...} &) (__f=...) at /usr/include/c++/11/bits/invoke.h:61
#7  0x0000555555571961 in std::__invoke_r<void, main()::<lambda(wayland::server::client_t&)>&, wayland::server::client_t&>(struct {...} &) (__fn=...) at /usr/include/c++/11/bits/invoke.h:111
#8  0x00005555555716c7 in std::_Function_handler<void(wayland::server::client_t&), main()::<lambda(wayland::server::client_t&)> >::_M_invoke(const std::_Any_data &, wayland::server::client_t &) (__functor=..., __args#0=...) at /usr/include/c++/11/bits/std_function.h:290
#9  0x0000555555576c63 in std::function<void (wayland::server::client_t&)>::operator()(wayland::server::client_t&) const (this=0x55555559d0b0, __args#0=...) at /usr/include/c++/11/bits/std_function.h:590
#10 0x0000555555572868 in wayland::server::display_t::client_created_func (listener=0x55555559d0f0, cl=0x55555559d6c0) at wayland-server.cpp:101
#11 0x00007ffff7f9c2f0 in ?? () from /lib/x86_64-linux-gnu/libwayland-server.so.0
#12 0x00007ffff7f9dd9c in wl_client_create () from /lib/x86_64-linux-gnu/libwayland-server.so.0
#13 0x00007ffff7f9deec in ?? () from /lib/x86_64-linux-gnu/libwayland-server.so.0
#14 0x00007ffff7f9bf4a in wl_event_loop_dispatch () from /lib/x86_64-linux-gnu/libwayland-server.so.0
#15 0x00007ffff7f9c725 in wl_display_run () from /lib/x86_64-linux-gnu/libwayland-server.so.0
#16 0x0000555555572f6c in wayland::server::display_t::run (this=0x7fffffffddc0) at wayland-server.cpp:224
#17 0x0000555555571271 in main () at lorie.cpp:21
A debugging session is active.

	Inferior 1 [process 1615551] will be killed.

Quit anyway? (y or n) [answered Y; input not from terminal]

```
